espnet2/asr_transducer/encoder/modules/convolution.py

```diff
@@ -107,3 +104,95 @@ def forward(
         x = self.pointwise_conv2(x).transpose(1, 2)
 
         return x, cache
+
+
+class ConvolutionalSpatialGatingUnit(torch.nn.Module):
+    """Convolutional Spatial Gating Unit module definition.
+
+    Args:
+        size: Initial size to determine the number of channels.
+        kernel_size: Size of the convolving kernel.
+        activation: Type of activation function.
+        norm_class: Normalization module class.
+        norm_args: Normalization module arguments.
+        dropout_rate: Dropout rate.
+        causal: Whether to use causal convolution (set to True if streaming).
+
+    """
+
+    def __init__(
+        self,
+        size: int,
+        kernel_size: int,
+        activation: torch.nn.Module = torch.nn.ReLU(),
```

Comment:
For CSGU, I recommend to set the `activation` as `Identity()`. Based on previous experiments, a nonlinear activation does not improve the performance.

Reply:
Removed activation argument, I manually set `Identity()` for now.